client/state/route/types.ts

```diff
@@ -0,0 +1,45 @@
+/**
+ * Internal dependencies
+ */
+import { ROUTE_CLEAR_LAST_NON_EDITOR, ROUTE_SET } from 'calypso/state/action-types';
+
+export type Query = {
+	[ key: string ]: string | number | undefined;
+	_timestamp?: number;
+};
+
+export type QueryState = {
+	initial: Query | false;
+	current: Query | false;
+	previous: Query | false;
+};
+
+export type Path = string;
```

Comment:
Type aliases like this are not very useful in TypeScript, where the type checking is structural rather than nominal. Simply using `string` everywhere is OK.